layerfetcher/layers_test.go

```diff
@@ -275,29 +281,32 @@ func TestPollLayerBlocks_AllHaveBlockData(t *testing.T) {
 	for i := 0; i < numPeers; i++ {
 		peer := p2pcrypto.NewRandomPubkey()
 		net.peers = append(net.peers, peer)
-		net.layerBlocks[peer] = generateLayerBlocks()
+		net.layerBlocks[peer] = generateLayerBlocks(i + 1)
 	}
-	l := NewMockLogic(net, db, &mockBlocks{}, &mockAtx{}, &mockFetcher{}, logtest.New(t))
+	l := NewMockLogic(net, db, &mockBlocks{}, &mockAtx{}, &mockFetcher{}, logtest.New(t, zapcore.DebugLevel))
 
 	layerID := types.NewLayerID(10)
 	res := <-l.PollLayerContent(context.TODO(), layerID)
 	assert.Nil(t, res.Err)
 	assert.Equal(t, layerID, res.Layer)
+
+	// should be the longest input vector from all peers
+	iv, err := db.GetLayerInputVectorByID(layerID)
+	assert.Nil(t, err)
+	assert.Equal(t, numPeers, len(iv))
 }
 
-func TestPollLayerBlocks_OneBlockNotFetched(t *testing.T) {
-	db := newLayerDBMock()
+func TestPollLayerBlocks_FetchBlockError(t *testing.T) {
 	net := newMockNet()
 	numPeers := 4
 	for i := 0; i < numPeers; i++ {
 		peer := p2pcrypto.NewRandomPubkey()
 		net.peers = append(net.peers, peer)
-		net.layerBlocks[peer] = generateLayerBlocks()
+		net.layerBlocks[peer] = generateLayerBlocks(i + 1)
 	}
-	l := NewMockLogic(net, db, &mockBlocks{}, &mockAtx{}, &mockFetcher{}, logtest.New(t))
+	l := NewMockLogic(net, newLayerDBMock(), &mockBlocks{}, &mockAtx{}, &mockFetcher{fetchError: ErrInternal}, logtest.New(t, zapcore.DebugLevel))
```

Comment:
why zapcore.DebugLevel is needed? usually TEST_LOG_LEVEL=debug env variable is sufficient. there was a couple of places where a particular log level was required for the test

Reply:
sorry. this is leftover from debugging that i forgot to remove.
with IDE it was easier for me to add here as needed. removed now.